### PR TITLE
XRT-2611: Sunspec 3.0 example

### PR DESCRIPTION
**Pull request by JonathanWhiteUK**

### CODE REVIEW

File: DeviceServices/virtual/commands/get_request.sh
Comment (dpazj) [inline DeviceServices/virtual/commands/get_request.sh:1]: I dont think you meant to update this?

File: Servers/opc-ua/federated/Explanation.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+## What does this show?
+
+This demonstration showcases the integration and interaction between Edge Connect Modbus Device Services and an OPC-UA server using MQTT. The focus is on monitoring and manipulating temperature data from battery cells. Although a simple demo running on one machine this is design to show how you can use XRT across multiple hardware nodes for example a gateway running a opc-ua server with multiple hardware nodes running device services sending data.
```
Comment (dpazj) [inline Servers/opc-ua/federated/Explanation.md]:
```suggestion
This demonstration showcases the integration and interaction between Edge Connect Modbus Device Services and an OPC-UA server using Sparkplug over MQTT. The focus is on monitoring and manipulating temperature data from battery cells. Although a simple demo running on one machine, this is designed to show how you can use XRT across multiple hardware nodes, for example a gateway running a opc-ua server with multiple hardware nodes running device services sending data.
```

File: Servers/opc-ua/federated/Explanation.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+Components:
+
+ -   Modbus device: Device to be read
+ -   Edge Connect Modbus Device Service: Responsible for    reading device data according to a specified schedule.
```
Comment (dpazj) [inline Servers/opc-ua/federated/Explanation.md]: extra spacing after 'for'

File: Servers/opc-ua/federated/README.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
+# Confederated OPC-UA Server and Modbus Device Service
+
+This example demonstrates how to operate the OPC-UA server using a OPC-UA nodeset and model to read a modbus device. The nodeset used is a
+[Sunspec model](https://sunspec.org/wp-content/uploads/2015/06/SunSpec-Information-Models-12041.pdf) that has been converted into an Nodeset.
```
Comment (dpazj) [inline Servers/opc-ua/federated/README.md]:
```suggestion
[Sunspec model](https://sunspec.org/wp-content/uploads/2015/06/SunSpec-Information-Models-12041.pdf) that has been converted to an OPC UA Nodeset.
```

File: Servers/opc-ua/federated/README.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+- Python 3 must also be installed. (can be checked using `python3 -v`)
+    
+## OPC-UA Browser Setup
+This section outlines the setup process for the IOTech OPC-UA browser. If you opt to use a different OPC-UA browser, please refer to the corresponding instructions specific to that browser.
+### Steps
+1. **Pull Docker Image**:
+    ```shell
+    docker pull iotechsys/opc-ua-browser:1.1
+    ```
+
```
Comment (dpazj) [inline Servers/opc-ua/federated/README.md:30]: extra whitespace

File: Servers/opc-ua/federated/README.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+[Sunspec model](https://sunspec.org/wp-content/uploads/2015/06/SunSpec-Information-Models-12041.pdf) that has been converted into an Nodeset.
+ Specifically, it utilizes a modified version of Sunspec's [Model805](https://github.com/sunspec/models/blob/master/json/model_805.json) to create an instance of a Lithium-ion module.
+
+The model in the OPC-UA server is a battery module model with three cells. This battery module will be simulated using a Modbus-tcp simulator where it will be read by a schedule of a XRT device service. This XRT device service sends over Sparkplug this data to the XRT OPC-UA Server where you the user can retrieve it using a OPC-UA Browser.
+
+For more information about the OPC-UA Server and Node Modelling please review the [OPC-UA Server](https://docs.iotechsys.com/edge-xrt30/server-components/opc-ua-server-component.html) documentation.
+
+For more information about the Modbus Device Service please review the [Modbus Device Service](https://docs.iotechsys.com/edge-xrt30/device-service-components/Modbus-device-service-component.html) documentation.
+
+### Prerequisites
```
Comment (dpazj) [inline Servers/opc-ua/federated/README.md]: leave a newline after a title - take a look at markdownlint vs code extension

File: Servers/opc-ua/federated/README.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+- Docker must be installed on your machine. (can be checked using `docker -v`)
+- XRT 3.0 must be installed, along with a copy of this repository (can be checked using `xrt` ). 
+- Python 3 must also be installed. (can be checked using `python3 -v`)
+    
+## OPC-UA Browser Setup
+This section outlines the setup process for the IOTech OPC-UA browser. If you opt to use a different OPC-UA browser, please refer to the corresponding instructions specific to that browser.
```
Comment (dpazj) [inline Servers/opc-ua/federated/README.md]: add a new line before a title

File: Servers/opc-ua/federated/README.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+   xrt ./deployment/config
+	```
+
+4. **Check OPC-UA Browser**:
+
+    Now if XRT is set up correctly you should see no error on the console output and be able to connect to the OPC-UA Server using the [OPC-UA Browser](http://0.0.0.0:8080/) and by entering your machines IP address into the 'Enter connection address' box (you can find this using `ip addr`).
+
+    When the Security pop up come up just click 'SAVE AND CONNECT' as for this example no security is set.
+
+    You should be able to see the nodeset model loaded in under:
```
Comment (dpazj) [inline Servers/opc-ua/federated/README.md]:
```suggestion
You should be able to see the model data defined by `opc-ua/deployment/models/Sunspec_Model805.json` loaded in under:
```

File: Servers/opc-ua/federated/README.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+   ```shell
+   xrt ./deployment/config
+	```
+
+
+4. **Seeing the results! :**
+
+   Now if we go onto our opc-ua browser we should see if we go into you should see this value incrementing by one every second:
+
+   modbus device -> Sparkplug -> xrt -> modbus-device1 -> cell_1_tmp
```
Comment (dpazj) [inline Servers/opc-ua/federated/README.md]: Should this just be
```suggestion
Sparkplug -> xrt -> modbus-device1 -> cell_1_tmp
```

File: Servers/opc-ua/federated/images/Deployment.png
Comment (dpazj) [inline Servers/opc-ua/federated/images/Deployment.png:1]: Can we remove then "Made with Whimsical" watermark?

File: Servers/opc-ua/federated/modbus-tcp/sim_files/MyScript
Comment (dpazj) [inline Servers/opc-ua/federated/modbus-tcp/sim_files/MyScript:1]: Should this be a .py file extension?

File: Servers/opc-ua/federated/opc-ua/deployment/config/logger.json [hunk 9]
```diff
@@ -0,0 +1,4 @@
+{
+  "Name": "logger",
+  "Level": "trace"
```
Comment (dpazj) [inline Servers/opc-ua/federated/opc-ua/deployment/config/logger.json]: Set to "info" for examples

File: Servers/opc-ua/federated/opc-ua/deployment/config/serverlogger.json [hunk 10]
```diff
@@ -0,0 +1,4 @@
+{
+  "Name": "serverlogger",
+  "Level": "trace"
```
Comment (dpazj) [inline Servers/opc-ua/federated/opc-ua/deployment/config/serverlogger.json]: Set to info for examples

File: Servers/opc-ua/federated/scripts/add_device.sh
Comment (dpazj) [inline Servers/opc-ua/federated/scripts/add_device.sh:1]: We can get rid of all the escapes by just wrapping the json message in ' ': 
```
#!/bin/sh

mosquitto_pub -t spBv1.0/iotech/REQUEST/xrt/modbus_tcp -m \
'{
  "client": "example",
  "request_id":"1010",
  "op": "device:add",
  }
}'
```

**Review by dpazj — CHANGES_REQUESTED** [review]
Can you replace references to OPC-UA with OPC UA

File: DeviceServices/virtual/commands/get_request.sh
Comment (dpazj) [inline DeviceServices/virtual/commands/get_request.sh:1]: Why has this been changed from the original?

File: Servers/opc-ua/federated/README.md [hunk 11]
```diff
@@ -0,0 +1,142 @@
+# Confederated OPC UA Server and Modbus Device Service
+
+This example demonstrates how to operate the OPC UA server using a OPC UA nodeset and model to read a modbus device. The nodeset used is a
+[Sunspec model](https://sunspec.org/wp-content/uploads/2015/06/SunSpec-Information-Models-12041.pdf) that has been converted to an OPC UA Nodeset.
+ Specifically, it utilizes a modified version of Sunspec's [Model805](https://github.com/sunspec/models/blob/master/json/model_805.json) to create an instance of a Lithium-ion module.
```
Comment (dpazj) [inline Servers/opc-ua/federated/README.md]: "it utilizes a modified version..."? This implies that we have modifed the original json model, which I believe is not the case?

Comment (JonathanWhiteUK) [reply Servers/opc-ua/federated/README.md]: changed this to 'it uses Sunspec'

File: DeviceServices/virtual/commands/get_request.sh
Comment (JonathanWhiteUK) [reply DeviceServices/virtual/commands/get_request.sh:1]: reverted change

**Review by dpazj — APPROVED** [review]
LGTM

**Pull request merged**